torchbenchmark/models/nvidia_deeprecommender/__init__.py

```diff
@@ -0,0 +1,100 @@
+# Benchmark created from NVidia DeepRecommender github project:
+#   https://github.com/NVIDIA/DeepRecommender
+#   a32a8a5c23092c551616acf6fac5b32e1155d18b
+# Test supports eval and train modes for cpu and cuda targets.
+#
+# Both nvtrain.py and nvinfer.py support all original command
+# line parameters but tensorflow dependency for logging has
+# been removed.
+
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import RECOMMENDATION
+
+import gc
+from .nvtrain import DeepRecommenderTrainBenchmark
+from .nvinfer import DeepRecommenderInferenceBenchmark
+
+class Model(BenchmarkModel):
+  
+  task = RECOMMENDATION.RECOMMENDATION
+  
+  def __init__(self, device="cpu", jit=False):
+    super().__init__()
+    self.device_name = device
+    self.not_implemented_reason = "Implemented"
+    self.eval_mode = True #default to inference
+
+    if jit:
+      self.not_implemented_reason = "Jit Not Supported"
+
+    elif self.device_name != "cpu" and self.device_name != "cuda":
+      self.not_implemented_reason = "device type not supported"
+
+    elif self.device_name == "cuda" and torch.cuda.is_available() == False:
```

Comment:
Since we are not supporting cuda on this model because of utilization issue, we should return "cuda is not available on this model" when `self.device_name == "cuda"`